lib/validator.js

```diff
@@ -15,48 +54,132 @@ var inMapping = {
+                // se we can't preinitialize it.
+                schema.required = schema.required || [];
+                if (schema.required.indexOf(inMapping[param.in]) < 0) {
+                    schema.required.push(inMapping[param.in]);
+                }
+            }
+
+            var reqPartSchema = schema.properties[inMapping[param.in]];
+            var paramSchema = { type: param.type };
+            supportedValidators.forEach(function(validator) {
+                paramSchema[validator] = param[validator];
+            });
+            reqPartSchema.properties[param.name] = paramSchema;
+            if (param.required) {
+                reqPartSchema.required = reqPartSchema.required || [];
+                reqPartSchema.required.push(param.name);
+            }
+        } else {
+            if (param.schema) {
+                schema.properties.body = param.schema;
+            } else {
+                schema.properties.body = {
+                    type: 'object'
+                };
+            }
+            if (param.required) {
+                schema.required = schema.required || [];
+                schema.required.push('body');
+            }
         }
     });
 
     return schema;
-}
+};
 
-var Validator = function(parameters) {
-    this.parameters = parameters;
-    this._validatorFunc = ajv.compile(convertToJsonSchema(parameters));
+/**
+ * Creates a function, that tries to coerce types of parameters in the
+ * incoming request, according to the provided parameters specification.
+ *
+ * @param parameters {Array} parameters swagger specification
+ * @returns {Function(req, HTTPError)} cersion function
+ * @private
+ */
+Validator.prototype._createTypeCoercionFunc = function(parameters) {
+    var code = '';
+    parameters.forEach(function(param) {
+        if (param.type === 'string') {
+            // Don't need to process strings
+            return;
+        }
+        var paramAccessor = 'req.' + inMapping[param.in] + '["' + param.name + '"]';
+        var paramCoercionCode;
+        var errorNotifier = 'throw new HTTPError({status:400,body:{type:"bad_request",'
+                + ' title:"Invalid parameters", detail: "data.'
+                + inMapping[param.in] + '.' + param.name + ' should be ' + param.type + '"}});\n';
+
+        switch (param.type) {
+            case 'integer':
+                paramCoercionCode = paramAccessor + ' = parseInt(' + paramAccessor + ');\n'
+                    + 'if (!Number.isInteger(' + paramAccessor + ')) {\n' + errorNotifier + '}\n';
+                break;
+            case 'number':
+                paramCoercionCode = paramAccessor + ' = parseFloat(' + paramAccessor + ');\n'
+                    + 'if (Number.isNaN(' + paramAccessor + ')) {\n' + errorNotifier + '}\n';
+                break;
+            case 'boolean':
+                paramCoercionCode = 'if(!/^true|false|1|0$/.test('
+                    + paramAccessor + '.toString())) {\n'
```

Comment:
Slightly safer might be to say paramAccessor + "''". That won't explode on methods that don't have a `.toString` method.